src/main/kotlin/CheckersModel.kt

```diff
@@ -1,85 +1,291 @@
 import kotlin.math.abs
+import kotlin.math.sign
 
-class CheckersModel : BaseModel() {
-    val board = CheckersBoard(8)
-    /* This functions checking turn for legacy and rules and also find checker, that will be eaten.
-    Its return value has 3 options:
-        if turn is illegal, it returns null,
-        if turn is legal but there aren't any checker will be eaten,
-            it returns square, from which the turn started
-        if turn is legal and some checker must be eaten,
-            it return square, where this checker is located */
+data class CheckersModel(val board : CheckersBoard = CheckersBoard(8)) : BaseModel() {
+    
+    constructor(model : CheckersModel) : this(CheckersBoard(model.board))  {
+        whoMoves = model.whoMoves
+        gameState = model.gameState
+        eatingChecker = model.eatingChecker
+        queenMovesInRow = model.queenMovesInRow
+    }
+
+    /**
+     * We divided one long turn with many eaten checkers into many turns with one eaten checker
+     * So we have to check that during one turn player moves only one checker
+     * @param eatingChecker is the checker that player moves during one turn
+     */
+    var eatingChecker : Square? = null
+
+    private var eatenList = ArrayList<Square>()
+
+    private var queenMovesInRow = 0
+    /**
+     * This functions checking turn for legacy and rules and also find checker, that will be eaten.
+         Its return value has 3 options:
+            if turn is illegal,
+                it returns null,
+            if turn is legal but there aren't any checker will be eaten,
+                it returns square, from which the turn started
+            if turn is legal and some checker must be eaten,
+                it return square, where this checker is located
+     * @param turn BaseTurn
+     * @return Square?
+     */
     override fun canMove(turn: BaseTurn): Square? {
+
         if (turn.playerColor != whoMoves) {
-            //player's color isn't correct
+            /**
+             * player's color isn't correct
+             */
```

Comment:
Sorry for that I haven't mentioned it, but documentations comments are used only before methods, classes and interfaces declarations. If you make a comment inside a method, use usual comments as you did before.

Reply:
Fixed